oshmem/mca/atomic/ucx/atomic_ucx_cswap.c

```diff
@@ -27,23 +27,23 @@ int mca_atomic_ucx_cswap(void *target,
                          int pe)
 {
     ucs_status_t status;
+    ucs_status_ptr_t status_ptr;
     spml_ucx_mkey_t *ucx_mkey;
     uint64_t rva;
+    uint64_t val;
+
+    if ((8 != nlong) && (4 != nlong)) {
+        ATOMIC_ERROR("[#%d] Type size must be 4 or 8 bytes.", my_pe);
+        return OSHMEM_ERROR;
+    }
 
     ucx_mkey = mca_spml_ucx_get_mkey(pe, target, (void *)&rva); 
+    val = (8 == nlong) ? *(uint64_t*)value : *(uint32_t*)value;
```

Comment:
can we unite this check with line 35?
e.g
if (nlong==8) val=.. else if(nlong==4) val=.. else error
(for fadd as well)